karma-dashboard.config.cjs

```diff
@@ -161,6 +162,25 @@ module.exports = function (config) {
     // how many browsers should be started simultaneously
     concurrency: Infinity,
 
+    // Sharding.
+    parallelOptions: {
+      shardStrategy: typeof config.shardIndex !== 'undefined' && typeof config.executors !== 'undefined' ? 'custom' : 'round-robin',
+      // If we're using custom sharding, just spin up 1 browser,
+      // but do the splitting in the custom strategy below.
+      executors: typeof config.shardIndex !== 'undefined' ? 1 : undefined, // Default is cpu cores - 1.
+      shardIndex: typeof config.shardIndex !== 'undefined' ? Number(config.shardIndex) : undefined,
+      // Re-implements a round-robin strategy, but with a custom shardIndex.
+      // Need to use the Function constructor here so we have access to config.shardIndex,
+      // because karma-parallel serializes this function.
+      customShardStrategy: new Function('parallelOptions', `
+        window.parallelDescribeCount = window.parallelDescribeCount || 0;
+        window.parallelDescribeCount++;
+        const shouldRunThisTest = (window.parallelDescribeCount % ${Number(config.executors)} === ${Number(config.shardIndex)});
+        console.log( "[SHARDING DEBUG]", "Count:", window.parallelDescribeCount, "Executors:", ${Number(config.executors)}, "Current shardIndex:", ${Number(config.shardIndex)}, "Should run test?", shouldRunThisTest);
+        return shouldRunThisTest;
+      `),
+    },
```

Comment:
will we be doing this with both the dashboard and the story editor? I was working in the story editor first

Reply:
If it works, we‘d want to do it for both, yes.

I was just experimentig with the dashboard first because there are fewer tests to run.

Reply:
Ah yup that makes sense. Thanks!